core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -236,22 +236,22 @@ class KafkaApis(val requestChannel: RequestChannel,
     if (isBrokerEpochStale(stopReplicaRequest.brokerEpoch)) {
       // When the broker restarts very quickly, it is possible for this broker to receive request intended
       // for its previous generation so the broker should skip the stale request.
-      info("Received stop replica request with broker epoch " +
+      info("Received StopReplica request with broker epoch " +
         s"${stopReplicaRequest.brokerEpoch} smaller than the current broker epoch ${controller.brokerEpoch}")
       sendResponseExemptThrottle(request, new StopReplicaResponse(new StopReplicaResponseData().setErrorCode(Errors.STALE_BROKER_EPOCH.code)))
     } else {
-      val (result, error) = replicaManager.stopReplicas(stopReplicaRequest)
+      val (result, error) = replicaManager.stopReplicas(request.context.correlationId, stopReplicaRequest)
```

Comment:
The return type is a bit awkward. As far as I can tell, the left side is just returning the deletion status, which is taken from the request. Was this an optimization in order to avoid another traversal of the request?

Reply:
That's correct. Now, we also need the leaderEpoch to pass it to the txnCoordinator so I will refactor this.

Reply:
I have refactored this as well. It is much better now.